x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/LastOverTime.java

```diff
@@ -109,14 +116,23 @@ public LastOverTime withFilter(Expression filter) {
 
     @Override
     public DataType dataType() {
-        return field().dataType();
+        var dataType = field().dataType();
+        if (dataType.isCounter()) {
+            return switch (dataType) {
+                case COUNTER_DOUBLE -> DataType.DOUBLE;
+                case COUNTER_INTEGER -> DataType.INTEGER;
+                case COUNTER_LONG -> DataType.LONG;
+                default -> throw new InvalidArgumentException("Received an unsupported counter type in LastOverTime");
+            };
+        }
+        return dataType;
     }
 
     @Override
     protected TypeResolution resolveType() {
         return isType(
             field(),
-            dt -> (dt.isNumeric() && dt != DataType.UNSIGNED_LONG) || dt == DataType.TSID_DATA_TYPE,
+            dt -> (dt.isNumeric() && dt != DataType.UNSIGNED_LONG) || dt == DataType.TSID_DATA_TYPE || dt.isCounter(),
```

Comment:
Nit: can we add a `noCounter` method to DataType, similar to `noText`? These type checks could be simpler if we call `noCounter` first.